### PR TITLE
feat: PolygonETH/USD feed added with historic data func

### PR DESCRIPTION
**Pull request by RyanPaulGannon**
Same as the other PR, but with the Pinia store now in use - much better!

I presume we can close the other PR and merge this one? Also this maybe solves #4, or do we leave this open until we can render the data? 

For future reference, is this the best way to do this sort of commit? I thought I would be able to use the branch I'd created initially, but obviously with the updates the two were quite different, so I figured a new branch from the update was the best/quickest option to submit this commit?

### CODE REVIEW

File: composables/price.ts [hunk 1]
```diff
@@ -52,13 +52,24 @@ const aggregatorV3InterfaceABI = [
...
+// Polygon ETH/USD Chainlink contract
+const polygonEthUsdAddress = '0x7bac85a8a13a4bcd8abb3eb7d6b4d632c5a57676'
 
 let ethUsdChainlinkContract: ethers.Contract
+let polygonEthUsdChainlinkContract: ethers.Contract
 
 export const useEthPriceFeed = async () => {
+  const provider = await useRpcProvider()
   if (!ethUsdChainlinkContract) {
-    const provider = await useRpcProvider()
```
Comment (toniengelhardt) [inline composables/price.ts:60]: Getting the provider should be inside the "if" clause bc it is only needed if the property doesn't exist yet, but I'll fix it after merge.

**Review by toniengelhardt — APPROVED** [review]
Very nice!

**Comment by toniengelhardt** [comment]
Great work, closing the other one. Yes, for now I think it is the best way. Bc the project is fresh, a lot of restructuring will probably happen in the beginning and it will be a bit chaotic. Once we figured out the basic structure and everything it will be more organized and less overlap.

For now we'll just have to pull updates frequently and potentially resolve some conflicts.

File: composables/price.ts (hunk 1 above)
Comment (toniengelhardt) [reply composables/price.ts:60]: No probs ✌🏽

**Comment by RyanPaulGannon** [comment]
Class, that's really helpful. I'm gonna spend some time getting to grips with that, because I had wondered how to go about it, really cool that! 🙌

**Comment by toniengelhardt** [comment]
I'm also not 100% sure how to go about it. 

My idea was more or less like this:
- Data feeds pull data periodically from API/chain and push it into the Pinia store.
- UI uses data from the store to display figures, charts, etc.
- Composables as shortcuts that wrap store access for common things like token price, gas price, etc. for convenience and not to repeat the code all the time.

**Pull request merged**